src/wkt/src/wrappers.rs

```diff
@@ -215,6 +221,47 @@ impl crate::message::Message for Int64Value {
     }
 }
 
+impl crate::message::Message for FloatValue {
+    fn typename() -> &'static str {
+        "type.googleapis.com/google.protobuf.FloatValue"
+    }
+    fn to_map(&self) -> Result<crate::message::Map, crate::AnyError>
+    where
+        Self: serde::ser::Serialize + Sized,
+    {
+        // Encode special strings, see https://protobuf.dev/programming-guides/json/.
+        if self.is_nan() {
+            return encode_string::<Self>("NaN".to_string());
+        }
+        if self.is_infinite() && self.is_sign_positive() {
+            return encode_string::<Self>("Infinity".to_string());
+        }
+        if self.is_infinite() && self.is_sign_negative() {
+            return encode_string::<Self>("-Infinity".to_string());
+        }
+
+        encode_value::<Self>(serde_json::json!(self))
+    }
+    fn from_map(map: &crate::message::Map) -> Result<Self, crate::AnyError>
+    where
+        Self: serde::de::DeserializeOwned,
+    {
+        let val = map
+            .get("value")
+            .ok_or_else(crate::message::missing_value_field)?;
+        match val {
+            // Decode special strings, see https://protobuf.dev/programming-guides/json/.
+            serde_json::Value::String(string) => match string.as_str() {
+                "Infinity" => Ok(f32::INFINITY),
+                "-Infinity" => Ok(f32::NEG_INFINITY),
+                "NaN" => Ok(f32::NAN),
+                _ => crate::message::from_value::<Self>(map),
+            },
+            _ => crate::message::from_value::<Self>(map),
+        }
```

Comment:
Would love a recommendation of a better way to write this. In this way I have main code path duplicated which I think could be improved.